### PR TITLE
Pluginise skip ad

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Moving this plugin out of the events file as I am going to start cleaning that up as GA is coming in, and it's a bit of a mess anyhow.
## What is the value of this and can you measure success?

I can eventually ad GA, simple, easy peasy.
## Does this affect other platforms - Amp, Apps, etc?

Not at all
## Screenshots

screen shot 2016-07-21 at 13 03 58
## Request for comment

@akash1810 @gidsg

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -273,7 +275,7 @@ define([
                                             }
                                         });
                                         player.on('adstart', function() {
-                                            player.adSkipCountdown(15);
+                                            player.skipAd(mediaType, 15);
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: what's `mediaType` for? are we planning on pre-rolling audio?

File: static/src/javascripts/projects/common/modules/media/videojs-plugins/skip-ad.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                        currentTime = adsManager.getCurrentAd().getDuration() - adsManager.getRemainingTime(),
+                        skipTime    = parseInt((skipTimeout - currentTime).toFixed(), 10);
+
+                    if (skipTime > 0) {
+                        $('.js-skip-remaining-time', this.el()).text(skipTime);
+                    } else {
+                        window.clearInterval(intervalId);
+                        $('.js-ads-skip', this.el())
+                            .html(
+                                '<button class="js-ads-skip-button vjs-ads-skip__button" data-link-name="Skip video advert">' +
```
Comment (akash1810) [inline static/src/javascripts/projects/common/modules/media/videojs-plugins/skip-ad.js]: is it worth pulling this into a file in its own right?

Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/media/videojs-plugins/skip-ad.js]: Trying to touch as little as possible - and this is all going to go away soon anyhow - so not worth the effort.

The point of this is to help clear out the events file.

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: It fires and event saying it's been skipped.
The event name, for now, are `media:{mediaType}:{eventType}`.

**Comment by akash1810** [comment]
👍

**Pull request merged**